pkg/config/product.go

```diff
@@ -35,17 +27,38 @@ type ProductSpec struct {
 	Properties map[string]interface{} `yaml:"properties"`
 }
 
+// KeyName returns a sanitized key name for the product.
+func (p *Product) KeyName() string {
+	// Replace any character that is not a letter, digit, or underscore with a
+	// single underscore.
+	reg := regexp.MustCompile(`[^a-zA-Z0-9_]+`)
+	key := reg.ReplaceAllString(p.Name, "_")
+
+	// Remove leading/trailing underscores that might result from the replacement.
+	key = strings.Trim(key, "_")
+
+	// Collapse multiple underscores into a single one.
+	key = regexp.MustCompile(`_+`).ReplaceAllString(key, "_")
+
+	// Ensure the name doesn't start with a digit by prefixing it with an
+	// underscore if it does.
+	if len(key) > 0 && '0' <= key[0] && key[0] <= '9' {
```

Comment:
Do we actually consider "" (empty string) to be a valid product name? Shouldn't it be an error?

Reply:
Empty string means it's not a Helm chart product. And, every product must have a associated Helm chart, so it will error out earlier if the configuration doesn't find a Helm chart for the product.